src/KPM/Lanczos.jl

```diff
@@ -104,10 +104,17 @@ function lanczos(mulA!, mulS!, v; min_iters, resolution=Inf, lhs=zeros(length(v)
     Sw  = zero(v)
 
     mulS!(Sv, v)
-    c = sqrt(real(dot(v, Sv)))
-    @assert c ≈ 1 "Initial vector not normalized"
-    @. v /= c
-    @. Sv /= c
+    vSv = dot(v, Sv)
+
+    # Check internal consistency of S and v. Valid under FP64 precision and
+    # assuming that S has characteristic scale 1.
+    atol = norm(v)^2 * length(v) * 1e-12
+    @assert isapprox(imag(vSv), 0; atol) # Hermitian S
```

Comment:
Just to make sure I'm following, the main reason this eliminates the Lanczos errors is because of a reasonable loosening of the tolerance?